src/lazy.rs

```diff
@@ -0,0 +1,89 @@
+use std::cell::UnsafeCell;
+use std::fmt;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+pub struct Lazy<T> {
+    val: UnsafeCell<Option<T>>,
+}
+
+pub struct AtomicLazy<T> {
+    val: UnsafeCell<Option<T>>,
+    state: AtomicUsize,
+}
+
+impl<T> Lazy<T> {
+    pub fn new() -> Lazy<T> {
+        Lazy { val: UnsafeCell::new(None) }
+    }
+
+    pub fn is_some(&self) -> bool {
+        self.as_ref().is_some()
+    }
```

Comment:
`Lazy` is used here https://github.com/carllerche/mio/blob/custom-readiness-queue/src/channel.rs#L96